File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Diagrams/getdiagramfontinfo.rb
module Asposediagramjava
  module GetDiagramFontInfo
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        fonts = diagram.getFonts()

        i = 0
        while i < fonts.getCount()
            font = fonts.get(i)
            # Display information about the fonts
            puts font.getName()

            i +=1
        end
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithDiagrams/GetDiagramFontInfo.php
<?php
namespace Aspose\Diagram\WorkingwithDiagrams;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class GetDiagramFontInfo{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram = new Diagram($dataDir."drawing.vsd");

        $fonts = $diagram->getFonts();

        $i = 0;
        while ($i<sizeof($fonts->getCount())) {
        $font = $fonts->get($i);
        # Display information about the fonts
        print $font->getName();
        $i+=1;
        }
    }
}